### DISCUSSION TITLE
Question about `headers_and_libraries` lecture

### DISCUSSION THREAD

**Discussion by alexkrz** [cpp-for-yourself/lectures-and-homeworks#102] [Issues withe lectures] [answered]
Hi Igor,

thank you for your great C++ educational videos!
After following the video on `headers_and_libraries`, I have two questions:

1.) How do you configure the Marp slides, such that the content does not exceed the slide borders?
When I preview the Markdown files in my VSCode and when I convert the Markdown files to pdf format, the content unfortunately does not fit on the default slide size.

2.) I cannot reproduce the example with the `ml.h`, `ml.cpp` and `predict_prices.cpp` files.
When running `c++ -std=c++17 predict_prices.cpp -I . -o predict_prices`, my compiler complains that the function `MagicallyGetBitcoinPrices()` is not defined.

I appreciate your input to these questions.

Cheers, Alex

**Comment by niosus** [comment, marked as answer]
Hey @alexkrz. Glad you like the videos!

As for your questions:
1. All of the configuration for Marp is stored here: https://github.com/cpp-for-yourself/lectures-and-homeworks/blob/main/theme.css 
That being said, I'm not an expert in this and it has been breaking a couple of times for me before. In addition to this, in VSCode I also allow HTML in the Marp slides. But that is about everything I can come up with. Hope it helps. In the later videos I went away from the idea of using Marp and slides overall 🤷 
2. As for reproducing what I have in slides, I did not write explicitly the `MagicallyGetBitcoinPrices` function out. You can write a very simple function that returns whatever vector of integers to play around with this. Does this answer your question?

**Reply by alexkrz** [reply]
Thank you for your quick reply!
With your help I got Marp to render the slides correclty.
I added the following lines to `.vscode/settings.json`:
```json
{
    "markdown.marp.themes": [
        "theme.css"
    ],
    "markdown.marp.enableHtml": true
}
```

For point 2.) I wrote a small function `MagicallyGetBitcoinPrices()` that returns a vector of integers. Now I get the expected error from the linker.
For me to follow along it was a bit frustrating that the example was incomplete due to space constraints.

I really enjoyed the example from the follow-up lecture on build-systems. This one worked like a charm.

Cheers!

**Reply by niosus** [reply]
Glad you figured everything out!

And thanks for your feedback! 🙏  It totally makes sense. I think I'll add the code for this function to the example too and try to keep the examples completely compilable in the future. 🤔
